renderer.cpp

```diff
@@ -61,29 +61,25 @@ void Renderer::renderDeferredSprites() {
     add(elem.d, elem.p.x, elem.k.y, elem);
   }
   if (!vertices.empty()) {
+    CHECK_OPENGL_ERROR();
     SDL::glBindTexture(GL_TEXTURE_2D, *currentTexture);
     SDL::glEnable(GL_TEXTURE_2D);
-    checkOpenglError();
     SDL::glEnableClientState(GL_VERTEX_ARRAY);
     SDL::glEnableClientState(GL_TEXTURE_COORD_ARRAY);
     SDL::glEnableClientState(GL_COLOR_ARRAY);
-    checkOpenglError();
     SDL::glColorPointer(4, GL_FLOAT, 0, colors.data());
-    checkOpenglError();
     SDL::glVertexPointer(2, GL_FLOAT, 0, vertices.data());
-    checkOpenglError();
     SDL::glTexCoordPointer(2, GL_FLOAT, 0, texCoords.data());
-    checkOpenglError();
     SDL::glDrawArrays(GL_TRIANGLES, 0, vertices.size() / 2);
-    checkOpenglError();
-    checkOpenglError();
     vertices.clear();
     texCoords.clear();
     colors.clear();
+
     SDL::glDisableClientState(GL_VERTEX_ARRAY);
     SDL::glDisableClientState(GL_TEXTURE_COORD_ARRAY);
     SDL::glDisableClientState(GL_COLOR_ARRAY);
     SDL::glDisable(GL_TEXTURE_2D);
+    CHECK_OPENGL_ERROR();
```

Comment:
Isn't it better to check for error more often?

Reply:
There is nothing wrong with this code. IMO GL_INVALID_ENUM that you mentioned was caused by  some gl function called before renderDeferredSprites.

Reply:
You can lookup each of these functions at this site:
https://www.khronos.org/registry/OpenGL-Refpages/gl2.1/xhtml/glEnableClientState.xml
It enumerates all situations in which an error can be reported.

Reply:
Generally glGetError() is not very useful. I put CHECK_OPENGL_ERROR at the beginning and the end of this block of code so that errors from one place won't show up in completely different place.